### ISSUE TITLE
Updating properties of the openstack integrator charm do not propagate automatically

### ISSUE THREAD

**Issue by CalvinHartwell** [juju-solutions/charm-openstack-integrator#10]
Hi all,

The open stack integrator charm works well, load-balancers and storage through cinder can be created. However, if you accidentally get the floating-network-id or subnet-id incorrect, they cannot be changed after deployment right now. If you attempt to update them by changing the charm config or by redeploying the bundle with updated values, they are ignored.

Set these values in the bundle: 

```
  openstack-integrator:
    charm: cs:~containers/openstack-integrator
    num_units: 1
    options:
      #endpoint-tls-ca: include-base64:///home/ubuntu/openstack-ca.crt
      password: <CATS-ARE-GOD>
      project-domain-name: admin_domain
      user-domain-name: admin_domain
      username: admin
      project-name: admin
      region: RegionOne
      auth-url: http://172.16.7.18:5000/v3
      subnet-id: 21ab2f91-9453-4047-a27d-b364a1d0593d
      floating-network-id: 06adf264-0048-4ef5-b2f2-1bbeaf48868d
      manage-security-groups: true
```

...and then change one of the values:

```
  openstack-integrator:
    charm: cs:~containers/openstack-integrator
    num_units: 1
    options:
      #endpoint-tls-ca: include-base64:///home/ubuntu/openstack-ca.crt
      password:  <CATS-ARE-GOD>
      project-domain-name: admin_domain
      user-domain-name: admin_domain
      username: admin
      project-name: admin
      region: RegionOne
      auth-url: http://172.16.7.18:5000/v3
      subnet-id: 21ab2f91-9453-4047-a27d-b364a1d0593d
      floating-network-id: kittensarecute
      manage-security-groups: true
```

I assume other properties probably can't be updated either, but we hit this issue with the floating-network-id. The others should also be tested, in case the user picks the wrong value. 

Many thanks!

**Comment by johnsca** [comment]
Hrm.  I thought this should have been fixed in #5.  Can you confirm that this issue is still occurring with cs:~containers/openstack-integrator-5 or later (current charm rev is 7)?

**Comment by dshcherb** [comment]
This when_not decorator is problematic because kubernetes-master.cloud.ready flag is not cleared when there are new changes propagated from the integrator charm (this should be relevant for any integrator):

https://github.com/juju-solutions/kubernetes/blob/cea46f6feceb838fd511bfff47c79260dc46b60a/cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py#L1784

What we need is a trigger that clears kubernetes-master.cloud.ready based on the fact that there are changes propagated by the integrator over the relation.
